core/src/main/java/io/ultrabrew/metrics/data/BasicHistogramAggregator.java

```diff
@@ -161,13 +161,18 @@ private Cursor newCursor(boolean sorted) {
     private int i = -1;
     private long base = 0;
     private int[] table;
+    private final Integer[] sortedRef;
 
     private CursorImpl(final String[][] tagSets, final boolean sorted) {
+      // Holds the indexes of tagSets in the case of sorted iteration
+      sortedRef = new Integer[tagSets.length];
+      for (int i = 0; i < tagSets.length; i++) {
+        sortedRef[i] = i;
+      }
+      this.tagSets = tagSets;
       if (sorted) {
-        this.tagSets = tagSets.clone();
-        Arrays.sort(this.tagSets, TagSetsHelper::compare);
-      } else {
-        this.tagSets = tagSets;
+        TagSetsComparator comparator = new TagSetsComparator(tagSets); 
```

Comment:
This comparator doesn't actually sort the `tagSets` but sorts the indices of `tagSets` ( through `sortedRef`, refer the logic at https://github.com/ultrabrew/metrics/pull/28/files#diff-fa8eef85c6812afee9ae6b074bdc7cbfR47-R48). This is done to avoid cloning the whole `tagSets` which is several orders magnitude expensive than the `sortedRef` array. So, a static method doesn't work in this case since I'm using `Arrays.sort` and that uses a comparator.

Reply:
Is it, because that is just creating one array of pointers to strings compared to creating an array of pointers and each pointer is again a new Integer object. I would not buy that argument without proof of performance

Reply:
Array.copy is a shallow copy